### ISSUE TITLE
Possible Memory leak after calling `read_block_from_disk`

### ISSUE THREAD

**Issue by ismaelsadeeq** [stickies-v/py-bitcoinkernel#9]
I was reading a large number of blocks from the data directory for a parallel task when I discovered that my script was getting killed after processing `~3000` blocks.

It seems there might be a memory leak occurring after calling `read_block_from_disk`. My suspect is that `kernel_block_destroy` is not being executed?. I plan to investigate this further but have not yet done so.

I slimmed my script for you to be able to reproduce this issue by running this branch https://github.com/ismaelsadeeq/py-blockparser/tree/12-2024-detect-memory-leak.

I started reading blocks from height `824147`, and the process was killed at height `827214`.

```bash
(.venv) root@vmi1649784:/block_parser# python3 src/main.py --datadir='/root/.bitcoin' --chain_type=mainnet --start_height=824147 --end_height=876041 
Memory usage at height: 824147 is 868 MB
Memory usage at height: 824148 is 868 MB
Memory usage at height: 824149 is 868 MB
Memory usage at height: 824150 is 868 MB

.......
Memory usage at height: 827212 is 11510 MB
Memory usage at height: 827213 is 11513 MB
Memory usage at height: 827214 is 11518 MB
Killed
```

 The memory usage spikes to ~11GB after processing ~3000 blocks.

**Comment by stickies-v** [comment]
Thanks a lot for finding and reporting this bug @ismaelsadeeq. #3 messed up the default pointer management behaviour. I fixed it in #9 - could you please let me know if this fixes it for you? I tested on signet and that seems to keep memory usage much more stable.

To test, update your pyproject.toml to use `"py-bitcoinkernel @ git+https://github.com/stickies-v/py-bitcoinkernel@309fe7025c74a32ef22051fa276a613f08572909",` as dependency and run `pip uninstall py-bitcoinkernel -y && pip install .`

**Comment by ismaelsadeeq** [comment]
> could you please let me know if this fixes it for you

Yes it does see https://github.com/stickies-v/py-bitcoinkernel/pull/10#pullrequestreview-2523433638
